tools/codegen/proc_interface_gen/template/ProcInterface.tmpl

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// THIS FILE IS AUTOMATICALLY GENERATED.
+
+package {{.PackageName}}
+
+import (
+{{range .Imports}}
+{{.}}
+{{end}}
+)
+
+
+type Instance struct {
+  {{range .ConstructorFields}}
+  {{.Name}} {{.Type.Name}}
+  {{end}}
+}
+
+type {{.Name}}Processor interface {
+  Configure{{.Name}}(types map[string]*Type) error
+  {{if .Check -}}
+    {{- .VarietyName}}{{.Name}}(instances map[string]*Instance) (bool, error)
```

Comment:
Gave it a shot. PTAL. I am sure we need a better location for the shared type 'CacheabilityInfo'. For now I have placed it within codegen/pkg/adapter